### DISCUSSION TITLE
How to make Mockoon recognize correct endpoint with query params from the request

### DISCUSSION THREAD

**Discussion by anapalmac** [mockoon/mockoon#839] [Help] [answered]
First of all, thank you for building this really cool tool. I'm enjoying exploring it! 

I did hit an issue that may block me from using it though. Mockoon seems to be using the wrong endpoint for a request.

I have the following 2 endpoints:
- GET /MyService/Clients/:id --> Returns the client with given Id. 
- GET /MyService/Clients/search  --> Requires query parameters (first and last name are required, plus some optional ones), returns client that matches the search criteria.

The issue is that when I try to call the second endpoint by using query parameters, Mockoon recognizes it as calling the first parameter with route param id = search. And therefore finds nothing.

Is there a way to make Mockoon recognize that I'm calling the second endpoint due to the fact that I'm using query string? Am I doing something wrong here?

![image](https://user-images.githubusercontent.com/110835674/197198123-f1015640-5ff5-4ebb-938f-80eab60a4800.png)

**Comment by 255kb** [comment, marked as answer]
Mockoon is using Express.js, so all of [Express' routing rules](https://expressjs.com/en/guide/routing.html) apply. 
If you define your routes in this order, the `:id` parameter will "intercept" the `/MyService/Clients/search` call and consider `search` to be the id. 
You need to reorder your routes by drag and dropping them, to put the more specific first:
```
GET /MyService/Clients/search
GET /MyService/Clients/:id
```

**Reply by anapalmac** [reply]
Thanks Guillaume! That did indeed solve my problem.
